### ISSUE TITLE
Allow access to multiple dns resources with a single allow access call

### ISSUE THREAD

**Issue by conectado** [CloudFire-LLC/cloudfire#2995]
~~Note that this not require any change on the portal since we can overload the `payload` field as we see fit.~~ Edit: If we want to do multiple resources at once, instead of just multiple domains, it requires an slight change in the allow access message to:
```
{
  client_id,
  resources: [
    resource,
    expires_at,
    payload
  ],
  reference
}
```
Note the multiple resources

and the reuse connection message to:

```
{
  gateway_id,
  resources: [
    resource_id,
    payload,
  ]
}
```

This is just an optimization(though an important one)

Needed due to #2994

**Comment by jamilbk** [comment]
Was this fixed in #5616 or #5575 ?

**Comment by conectado** [comment]
Deprecated by new control protocol
